### PR TITLE
Issue #25 - adding more meaningful ObjectMetadata to MkOcket

### PR DESCRIPTION
**Pull request by prondzyn**
Prepared instead of closed #36 pull request. Travis CI build passed now.

### CODE REVIEW

**Comment by prondzyn** [comment]
@caarlos0 here you have pull request with fixed Travis build. CC: @dmarkov

**Comment by dmarkov** [comment]
@prondzyn Let me find a reviewer for this pull request, thanks for submitting it

**Comment by dmarkov** [comment]
@pinaf please review this PR

**Comment by caarlos0** [comment]
@prondzyn why did you close #36 ?

**Comment by prondzyn** [comment]
@caarlos0 because #37 contains #36 plus Travis build fix so I thought you will be automatically assigned to #37. I understand I was wrong.

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * @throws Exception If fails
+     */
+    @Test
+    public void readContentTypeAndLengthFromMetadata() throws Exception {
+        final Region region = new MkRegion(this.temp.newFolder());
+        final Bucket bucket = region.bucket("test");
+        final Ocket ocket = bucket.ocket("hello.txt");
+        final String text = "hello, world!";
+        new Ocket.Text(ocket).write(text);
+        final Ocket readOcket = new Ocket.Text(bucket.ocket(ocket.key()));
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn let's rename this ocket to `read` and the other ocket to `write`.

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        final ObjectMetadata metadata = readOcket.meta();
+        Assert.assertEquals("text/plain", metadata.getContentType());
+        Assert.assertEquals(text.length(), metadata.getContentLength());
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn no test for the new `Headers.DATE` key ?

**Comment by pinaf** [comment]
@prondzyn 2 comments above

**Comment by pinaf** [comment]
@prondzyn what is the purpose of the MIME file? Is it used by the s3 library?

**Comment by prondzyn** [comment]
@pinaf 

I changed the names and added test to get Date from metadata.

The mime.types file is used by MimetypesFileTypeMap in MkOcket when no mime types found in standard system directories. Absence of this file broke the Travis [build 122](https://travis-ci.org/jcabi/jcabi-s3/builds/81857432). I added comment to the mime.types file to give a hint where it is used.

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    /**
+     * Date can be read from MkOcket metadata.
+     * @throws Exception If fails
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
+        final ObjectMetadata metadata = read.meta();
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn let's inline this variable

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        Assert.assertEquals(text.length(), metadata.getContentLength());
+    }
+
+    /**
+     * Date can be read from MkOcket metadata.
+     * @throws Exception If fails
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn let's inline this variable

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        final String text = "hello, world!";
+        new Ocket.Text(this.write).write(text);
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
+        final ObjectMetadata metadata = read.meta();
+        Assert.assertEquals("text/plain", metadata.getContentType());
+        Assert.assertEquals(text.length(), metadata.getContentLength());
+    }
+
+    /**
+     * Date can be read from MkOcket metadata.
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn `MkOcket can read date from metadata`

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+     * @throws Exception If fails
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn `readsDateFromMetadata`

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+        this.bucket = region.bucket("test");
+        this.write = this.bucket.ocket("hello.txt");
+    }
+
+    /**
+     * Content Type and Length can be read from MkOcket metadata.
+     * @throws Exception If fails
+     */
+    @Test
+    public void readContentTypeAndLengthFromMetadata() throws Exception {
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn `readsContentTypeAndLengthFromMetadata`

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    @Before
+    public void setUp() throws Exception {
+        final Region region = new MkRegion(this.temp.newFolder());
+        this.bucket = region.bucket("test");
+        this.write = this.bucket.ocket("hello.txt");
+    }
+
+    /**
+     * Content Type and Length can be read from MkOcket metadata.
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn `MkOcket can read Content Type and Length from metadata`

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+     */
+    private transient Ocket write;
+
+    /**
+     * Sets up common part of tests.
+     * @throws Exception If fails
+     */
+    @Before
+    public void setUp() throws Exception {
+        final Region region = new MkRegion(this.temp.newFolder());
```
Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn let's inline this variable

**Comment by pinaf** [comment]
@prondzyn thanks

**Comment by pinaf** [comment]
@prondzyn a few more minor comments and we are good to merge

File: src/test/java/com/jcabi/s3/mock/MkOcketTest.java (hunk 3 above)
Comment (prondzyn) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @pinaf do you think about such solution?
final ObjectMetadata metadata =
            new Ocket.Text(this.bucket.ocket(this.write.key())).meta();

Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn The first option works. Another one would be this:
```java
final ObjectMetadata metadata = new Ocket.Text(
   this.bucket.ocket(this.write.key())
).meta();
```

Comment (prondzyn) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @pinaf your looks better. Thanks!

Comment (pinaf) [inline src/test/java/com/jcabi/s3/mock/MkOcketTest.java]: @prondzyn there is a chance qulice complains (not sure if that bug was resolved it) so you might have to do this at the end - yeah, it looks weird :)
```java
final ObjectMetadata metadata = new Ocket.Text(
   this.bucket.ocket(this.write.key())
)
    .meta();
```

**Comment by prondzyn** [comment]
@pinaf code cleaned up

**Comment by pinaf** [comment]
@prondzyn excellent! thanks!

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor try to merge pls

**Comment by dmarkov** [comment]
@pinaf **21 mins** sent to your balance (ID `AP-1TR56345YP2041344`), many thanks! It took 90 hours and 23 mins.... extra minutes for review comments (c=6)... +21 added to your rating, at the moment it is: [+4217](http://www.netbout.com/b/35128?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy now pls

**Pull request merged**